src/objects/Points.js

```diff
@@ -144,13 +144,14 @@ function testPoint( point, index, localThresholdSq, matrixWorld, raycaster, inte
 		_ray.closestPointToPoint( point, intersectPoint );
 		intersectPoint.applyMatrix4( matrixWorld );
 
-		const distance = raycaster.ray.origin.distanceTo( intersectPoint );
+		const intersectPoint2Ray = raycaster.ray.origin.distanceTo( intersectPoint );
 
-		if ( distance < raycaster.near || distance > raycaster.far ) return;
+		if ( intersectPoint2Ray < raycaster.near || intersectPoint2Ray > raycaster.far ) return;
 
-		intersects.push( {
+		const distance2Ray = raycaster.ray.distanceToPoint( point );
 
-			distance: distance,
+		intersects.push( {
+			distance: distance2Ray,
```

Comment:
I don't think it's appropriate to change the `distance` value in this way since it will affect all existing applications that rely on the current implementation. I also don't think the new value can be considered as "more correct".

You can easily a custom distance value on app level so it's probably best to not change `testPoint()` at all.

Reply:
Yeah the implementation proposed will produce an incorrect result. `Ray.distanceToPoint` changes depending on whether the ray length. I think it's best to ask at the forum for help at the forum if the OP is having trouble retrieving closest points.